wasm-rpc-stubgen/src/stub.rs

```diff
@@ -332,37 +332,47 @@ fn collect_stub_interfaces(resolve: &Resolve, world: &World) -> anyhow::Result<V
 fn collect_stub_functions<'a>(
     functions: impl Iterator<Item = &'a Function>,
 ) -> anyhow::Result<Vec<FunctionStub>> {
-    Ok(functions
-        .map(|f| {
-            let mut params = Vec::new();
-            for (name, typ) in &f.params {
-                params.push(FunctionParamStub {
-                    name: name.clone(),
-                    typ: *typ,
-                });
-            }
+    let mut function_stubs = vec![];
+
+    for f in functions {
+        let mut params = Vec::new();
+        for (name, typ) in &f.params {
+            params.push(FunctionParamStub {
+                name: name.clone(),
+                typ: *typ,
+            });
+        }
 
-            let results = match &f.results {
-                Results::Named(params) => {
-                    let mut param_stubs = Vec::new();
-                    for (name, typ) in params {
-                        param_stubs.push(FunctionParamStub {
-                            name: name.clone(),
-                            typ: *typ,
-                        });
-                    }
-                    FunctionResultStub::Multi(param_stubs)
+        let results = match &f.results {
+            Results::Named(params) => {
+                let mut param_stubs = Vec::new();
+                for (name, typ) in params {
+                    param_stubs.push(FunctionParamStub {
+                        name: name.clone(),
+                        typ: *typ,
+                    });
                 }
-                Results::Anon(single) => FunctionResultStub::Single(*single),
-            };
-
-            FunctionStub {
-                name: f.name.clone(),
-                params,
-                results,
+                FunctionResultStub::Multi(param_stubs)
             }
-        })
-        .collect())
+            Results::Anon(single) => FunctionResultStub::Single(*single),
+        };
+
+        if results.is_empty() {
+            function_stubs.push(FunctionStub {
+                name: format!("blocking-{}", f.name),
+                params: params.clone(),
+                results: results.clone(),
+            })
```

Comment:
if results is empty, then there will be 2 functions ?

1. blocking-function_name
2. function_name 
?